vamb/__main__.py

```diff
@@ -1442,7 +1447,41 @@ def run_bin_default(opt: BinDefaultOptions):
         FastaOutput.try_from_common(opt.common),
         None,
     )
-    del latent
+
+
+class RunDefault:
+    pass
+
+
+class RunTrain:
+    pass
+
+
+class RunCluster:
+    def __init__(self, latent: Path):
+        self.latent = latent
+
+
+def load_train_bin(
+    opt: BinDefaultOptions, partial_mode: Union[RunDefault, RunTrain, RunCluster]
+):
+    composition, abundance = load_composition_and_abundance(
+        vamb_options=opt.common.general,
+        comp_options=opt.common.comp,
+        abundance_options=opt.common.abundance,
+        binsplitter=opt.common.output.binsplitter,
+    )
+
+    latent = None
+
+    if isinstance(partial_mode, (RunDefault, RunTrain)):
+        latent = run_train_vae(opt, composition, abundance)
+
+    if isinstance(partial_mode, RunCluster):
+        latent = partial_mode.latent
+
+    if isinstance(partial_mode, (RunDefault, RunCluster)):
+        run_cluster_and_write_files(latent, opt, composition)
 
```

Comment:
This function needs to make sure it saves the computed results, i.e. if `RunTrain` is run, then the resulting `latent` is saved as `latent.npz` in the appropriate file.